### DISCUSSION TITLE
An error occurred while customizing the node location

### DISCUSSION THREAD

**Discussion by cuicw** [GazzolaLab/PyElastica#327] [Q&A] [answered]
Hello，I 'm trying to build a catenary configuration bar as shown in the following figure. I 've discretized the catenary by calculation and wanted to import it directly into straight _ rod, but the following error occurred
AssertionError: 
Not equal to tolerance rtol=1e-07, atol=2.22045e-12
 Rod normal and tangent are not perpendicular to each other!
Mismatched elements: 2002 / 2002 (100%)
Max absolute difference: 0.99777105
Max relative difference: inf
 x: array([0.061433, 0.061463, 0.061494, ..., 0.997261, 0.996986, 0.996699])
 y: array(0)

The following figure is my code details, and the initial catenary configuration.
![9e73cc3a31f706ce3df36529140c717](https://github.com/GazzolaLab/PyElastica/assets/154817929/ee71e73a-7c4f-425e-8acc-dcb1d1ae25a7)

**Comment by armantekinalp** [comment]
@Ali-7800

**Comment by Ali-7800** [comment, marked as answer]
@cuicw You need to specify the the directors for the rods, this is a way of finding them:
```
import numpy as np
from elastica._linalg import (
    _batch_norm,
)

director_collection = np.zeros((3, 3, n_elem))
normal_collection = np.zeros((3, n_elem))
binormal_collection = np.zeros((3, n_elem))

# Compute rod tangents using positions
position_for_difference = position_collection
position_diff = position_for_difference[..., 1:] - position_for_difference[..., :-1]
rest_lengths = _batch_norm(position_diff)
tangents = position_diff / rest_lengths

for i in range(n_elem):
    # Compute the normal vector at each element. Since we allow helix radius to vary, we need to compute
    # vectors creating normal for each element.
    normal_collection[0, i] = -tangents[1, i]
    normal_collection[1, i] = tangents[0, i]
    normal_collection[..., i] /= np.linalg.norm(normal_collection[..., i])

    binormal_collection[..., i] = np.cross(tangents[..., i], normal_collection[..., i])
    director_collection[..., i] = np.vstack(
        (
            normal_collection[..., i],
            binormal_collection[..., i],
            tangents[..., i],
        )
    )
```

Then add `directors=director_collection,` to the rod initialization.

**Reply by cuicw** [reply]
Thank you very much.

**Reply by cuicw** [reply]
The problem has been solved.

**Comment by Ali-7800** [comment]
Let us know if you encounter any other issues, we will close this discussion.
